web/streetmanager/srwr_process.php

```diff
@@ -0,0 +1,345 @@
+		  $printrow[$activityId]['permit_status'] = $row[29] == '1' ? 'Accepted' : 'Unknown permit';
+	   }
+	 } else if ($row[1] == '099') {
+	   $companies[ $row[3] . $row[4] ] = $row[5];
+	 }
+}
+
+function manipulateData() {
+	global $printrow;
+
+	foreach ($printrow as $key => $code) {
+		checkDates($key, $code);
+		addCompanies($key, $code);
+		addMissingFields($key);
+	}
+
+}
+
+// Put in defaults for any fields that may not have been in the data, but the database
+// will be expecting
+
+function addMissingFields($key) {
+	global $printrow;
+
+	if (!array_key_exists('work_category', $printrow[$key])) {
+		$printrow[$key]['work_category'] = _workCategory('00');
+	}
+
+	if (!array_key_exists('works_location_type', $printrow[$key])) {
+		$printrow[$key]['works_location_type'] = _siteLocations('0');
+	}
+
+	if (!array_key_exists('project_name', $printrow[ $key ]) ) {
+		$printrow[ $key ]['project_name'] = '';
+	}
+
+	if (!array_key_exists('traffic_management_type', $printrow[ $key ]) ) {
+		$printrow[ $key ]['traffic_management_type'] = _trafficManagementType('01');
+	}
+
+	if (!array_key_exists('close_footway', $printrow[ $key ]) ) {
+		$printrow[ $key ]['close_footway'] = 'Not known';
+	}
+
+	if (!array_key_exists('work_status', $printrow[ $key ]) ) {
+		$printrow[ $key ]['work_status'] = _activityStatuses('');
+	}
+
+	if (!array_key_exists('permit_status', $printrow[ $key ]) ) {
+		$printrow[ $key ]['permit_status'] = 'Unknown permit';
+	}
+
+}
+
+// If there's a start date and no end date or vice versa, assume they are the same.
+// Mark for deletion if there are neither.
+// If the end date is older than today also mark for deletion
+
+function checkDates($key, $code) {
+	global $printrow;
+
+	if (array_key_exists('proposed_start_date', $code) && !array_key_exists('proposed_end_date', $code)) {
+		$printrow[$key]['proposed_end_date'] = $printrow[$key]['proposed_start_date'];
+	} else if (array_key_exists('proposed_end_date', $code) && !array_key_exists('proposed_start_date', $code)) {
+		$printrow[$key]['proposed_start_date'] = $printrow[$key]['proposed_end_date'];
+	}
+
+	if (!array_key_exists('proposed_start_date', $printrow[$key])) {
+		$printrow[$key]['drop_data'] = 1;
+	} else if (date_create($printrow[$key]['proposed_end_date'])->format("Y-m-d") < date_create('now')->format("Y-m-d")) {
+		$printrow[$key]['drop_data'] = 1;
+	}
+}
+
+// Look up the companies by code, all in the supplied csv data
+
+function addCompanies($key, $code) {
+	global $printrow;
+	global $companies;
+
+	if (array_key_exists('project_ref', $code) ) {
+		$company_key = $code['project_ref'];
+		if ( array_key_exists($company_key, $companies) ) {
+			$printrow[ $key ]['project_name'] = $companies[$company_key];
+		}
+	}
+}
+
+function insertRows() {
+	global $printrow;
+
+	$query = createStreetManagerQuery();
+	foreach ($printrow as $key => $object) {
+		$object['permit_reference_number'] = $key;
```

Comment:
What do these keys look like? Are they comparable to street manager? If we wanted to remove all entries added from this script, would we be able to do so?